mobile/test/core/data/classroom_database_test.dart

```diff
@@ -28,31 +56,112 @@ Future<void> main() {
   Database database;
   VmDatabase vmDatabase;
 
-  setUp(() async {
+  Future connectDatabase() async {
     vmDatabase = VmDatabase.memory(setup: (db) {
       db.execute('PRAGMA foreign_keys = ON');
     });
     database = Database(vmDatabase);
-    database.into(database.userModels).insert(tUserCompanion);
+    await database.into(database.userModels).insert(tUserCompanion);
+  }
+
+  group("insertClassroom", () {
+    setUp(() async {
+      await connectDatabase();
+    });
+
+    tearDown(() async {
```

Comment:
The `tearDown` method is being called on every group, shouldn't this be on the outer `setUp`?